hstream/src/HStream/Server/Core/Stream.hs

```diff
@@ -222,13 +222,16 @@ data Rid = Rid
   { rShardId  :: Word64
   , rBatchId  :: Word64
   , rBatchIdx :: Word32
-  } deriving (Show, Eq)
+  } deriving (Eq)
 
 instance Ord Rid where
-  Rid{rShardId=rs1, rBatchIdx=rb1} <= Rid{rShardId=rs2, rBatchIdx=rb2}
+  Rid{rShardId=rs1, rBatchId=rb1} <= Rid{rShardId=rs2, rBatchId=rb2}
     | rs1 /= rs2 = rs1 <= rs2
     | otherwise = rb1 <= rb2
```

Comment:
Does this mean  `Rid 100 100 0 <= Rid 100 100 1` and `Rid 100 100 0 /= Rid 100 100 1` are True? Which means `Rid 100 100 0 < Rid 100 100 1`? Is this your expectation?

Reply:
Only care about `shardId` and `batchId` here, which means:
- `Rid 100 100 100 <= Rid 101 101 101`: compare shardId first
- `Rid 100 100 0 <= Rid 100 101 1`: compare batchId if shardId is same
- `Rid 100 100 99 <= Rid 100 100 1`: ignore batchIndex

Reply:
If you do not care about `rBatchIdx`, then you should implement your own `Eq` instance, not use deriving.

Which is  `Rid 100 100 _ === Rid 100 100 _`